modules/gateway/peersmanager.go

```diff
@@ -60,6 +60,25 @@ func (g *Gateway) numOutboundPeers() (numOutboundPeers int) {
 	return numOutboundPeers
 }
 
+// randomPotentialPeer returns a random node from the gateway. Unlike
+// randomNode, it is not truly random because it prioritizes some nodes over
+// others. An error can be returned if there are no nodes in the node list.
+func (g *Gateway) randomPotentialPeer() (modules.NetAddress, error) {
+	if len(g.nodes) == 0 {
+		return "", errNoPeers
+	}
+
+	// Try three times to select a prioritized node.
+	for i := 0; i < 3; i++ {
+		addr, _ := g.randomNode()
+		if g.nodes[addr].Prioritize {
+			return addr, nil
+		}
+	}
```

Comment:
I think it would be far more secure to scan the whole list and specifically pull out the outbound ones, and then choose randomly from those. If we have connectable outbound peers, we should reuse them, and only fall back to our inbound peers if all remaining outbound connections have failed but we still need more peers.

Outbound is likely to be a tiny percentage of the list, almost worth putting in a separate place in memory.